### PR TITLE
AHRS: Improve accelerometer noise handling

### PR DESCRIPTION
**Pull request by niewim19**
## Description

Multiple commits, all but the first ([devekf: add selection to print versors or attitude](https://github.com/phoenix-pilot/phoenix-pilot-core/commit/dd538f86337b874e910987409eff4a1bdf628bc2)) were clang-format-ed before pushing.

Changes regarding minimizing accelerometer measurement noise impact on attitude estimation. Main changes were:
 - add acceleration logging possibilities to test applications
 - use ekf filtered accelerations in attitude estimation instead of basing attitude estimation on raw accelerometer measurements
 - heavily increased accelerometer noise factor in ekf R matrix
 - change in quadcontrol pid calculations so that they do not use error rate derrivative, but angular rates from ekf
 - small naming fixes

## Motivation and Context

High levels of noise were found in signals on which attitude estimation is based on. Noise in such levels (stdev of noise > 4000 mm/s^2) was not encountered before as no props were mounted to the craft.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## How Has This Been Tested?

- [ ] Already covered by automatic testing.
- [ ] New test added: (add PR link here).
- [ ] Tested by hand on: zturn.

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing linter checks and tests passed.
- [ ] My changes generate no new compilation warnings for any of the targets.

## Special treatment

- [ ] This PR needs additional PRs to work (list the PRs, preferably in merge-order).
- [ ] I will merge this PR by myself when appropriate.

### CODE REVIEW

File: ekf/tests/sensortest.c [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+
+static int sensortest_motorsProcedure(void)
+{
+	const char pwmInit[] = "100000";
+
+	if (sensortest_allWrite(pwmInit, sizeof(pwmInit)) < 0) {
+		sensortest_emergencyShutdown(pwmInit, sizeof(pwmInit));
+		return -1;
+	}
+	sleep(9);
```
Comment (HBuczynski) [inline ekf/tests/sensortest.c:118]: Please add space between `sleep` and condition.

File: ekf/tests/sensortest.c [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+		return EXIT_FAILURE;
+	}
+
+	power = atoi(argv[1]);
+	if (power > THROTTLE_MAX) {
+		printf("Throtle of %u too big! Max throttle = %d%%", power, THROTTLE_MAX);
+		return EXIT_FAILURE;
+	}
+	if (power < THROTTLE_MIN) {
+		printf("Throtle of %d too low, Min throttle = %d%%\n", power, THROTTLE_MIN);
```
Comment (HBuczynski) [inline ekf/tests/sensortest.c:148]:
```suggestion
		printf("Throtle of %u too low, Min throttle = %d%%\n", power, THROTTLE_MIN);
```

File: ekf/tests/sensortest.c [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+		return -1;
+	}
+
+	return 0;
+}
+
+
+int main(int argc, char **argv)
+{
+	unsigned int power, i, flag = 0, ret;
```
Comment (HBuczynski) [inline ekf/tests/sensortest.c:135]:
```suggestion
	unsigned int throttle, i, flag = 0, ret;
```
To be compliant with our convention.

File: ekf/tests/sensortest.c [hunk 4]
```diff
@@ -0,0 +1,189 @@
...
+		}
+	}
+	if (flag != 0) {
+		sensortest_allClose();
+		return EXIT_FAILURE;
+	}
+
+	beginthread(sensortest_thread, 4, sensortest_common.stack, sizeof(sensortest_common.stack), NULL);
+
+	ret = sensortest_motorsProcedure();
```
Comment (HBuczynski) [inline ekf/tests/sensortest.c:181]: Function returns `int` type but `ret` is declared as `unsigned int`.

File: ekf/tests/sensortest.c [hunk 5]
```diff
@@ -0,0 +1,189 @@
...
+}
+
+static void sensortest_thread(void *arg)
+{
+	sensor_event_t aEvt, gEvt, mEvt;
+	ekf_state_t uavState;
+
+	if (ekf_init() == 0) {
+		ekf_run();
+	}
```
Comment (HBuczynski) [inline ekf/tests/sensortest.c:76]: Initialization of external modules should be placed outside thread function.

File: quadcontrol/control.c [hunk 6]
```diff
@@ -55,7 +55,11 @@ struct {
 } quad_common;
 
 
-enum { pwm_alt = 0, pwm_roll, pwm_pitch, pwm_yaw, pwm_max };
+enum { pwm_alt = 0,
+	pwm_roll,
+	pwm_pitch,
+	pwm_yaw,
+	pwm_max };
```
[the remark this answers is not in the document]
Comment (niewim19) [reply quadcontrol/control.c:62]: Me neither. However out clang-format does, and fails if enums are in one line

[the remark this answers is not in the document]
Comment (niewim19) [reply quadcontrol/control.c:62]: Going back to one line and ignore the clang output about it

File: ekf/sensc.c [hunk 7]
```diff
@@ -26,9 +26,7 @@
 
 #define SENSORHUB_PIPES 3 /* number of connections with sensorhub */
 
-typedef enum { fd_imuId = 0,
-	fd_baroId,
-	fd_gpsId } fd_id_t;
+typedef enum { fd_imuId = 0, fd_baroId, fd_gpsId } fd_id_t;
```
[the remark this answers is not in the document]
Comment (niewim19) [reply ekf/sensc.c:29]: Ignoring this remark, as it does not fit our coding convention

File: ekf/tests/devekf.c [hunk 8]
```diff
@@ -15,40 +15,73 @@
...
+
 int main(int argc, char **argv)
 {
 	ekf_state_t uavState;
-	quat_t q;
-	vec_t start = { 0 };
-	int i = 0;
+	enum printMode mode = prntVersor;
+
+	if (argc > 1) {
```
Comment (HBuczynski) [inline ekf/tests/devekf.c:59]: Do you expect only one argument? If so, then:
```suggestion
	if (argc != 2) {
	    /* handle error */
	}
```

File: ekf/tests/devekf.c [hunk 9]
```diff
@@ -15,40 +15,73 @@
...
+		if (atoi(argv[1]) == 1) {
+			mode = prntAtt;
+		}
+		if (atoi(argv[1]) == 2) {
+			mode = printAcc;
+		}
+	}
 
 	if (ekf_init() == 0) {
```
Comment (HBuczynski) [inline ekf/tests/devekf.c:68]: You don't handle situation when `ekf_init()` returns with error.

File: ekf/tests/sensortest.c [hunk 10]
```diff
@@ -0,0 +1,204 @@
...
+{
+	printf("Usage: %s [throttle]\n", appName);
+	printf("  throttle - integer in range [%d,%d], interpreted as percents\n", THROTTLE_MIN, THROTTLE_MAX);
+}
+
+
+static int sensortest_motorsWrite(const char *val, int sz)
+{
+	int i;
+	for (i = 0; i < NUM_OF_MOTORS; i++) {
```
Comment (HBuczynski) [inline ekf/tests/sensortest.c:49]: add space

File: ekf/tests/sensortest.c [hunk 11]
```diff
@@ -0,0 +1,201 @@
...
+	return 0;
+}
+
+
+int main(int argc, char **argv)
+{
+	unsigned int throttle, i, flag = 0;
+	int ret;
+
+	if (argc != 1) {
```
Comment (HBuczynski) [inline ekf/tests/sensortest.c:139]:
```suggestion
	if (argc != 2) {
```

File: ekf/tests/sensortest.c [hunk 12]
```diff
@@ -0,0 +1,201 @@
...
+	}
+
+	if (ekf_init() == 0) {
+		beginthread(sensortest_thread, 4, sensortest_common.stack, sizeof(sensortest_common.stack), NULL);
+	}
+	else {
+		printf("Cannot initialize ekf\n");
+		sensortest_motorsClose();
+		return EXIT_FAILURE;
+	}
```
Comment (HBuczynski) [inline ekf/tests/sensortest.c:191]:
```suggestion
	if (ekf_init() != 0) {
		printf("Cannot initialize ekf\n");
		sensortest_motorsClose();
		return EXIT_FAILURE;
	}
	
	beginthread(sensortest_thread, 4, sensortest_common.stack, sizeof(sensortest_common.stack), NULL);
```

**Comment by niewim19** [comment]
Sensortest and devekf were tested after changes and both work good

**Review by HBuczynski — APPROVED** [review]

**Pull request merged**